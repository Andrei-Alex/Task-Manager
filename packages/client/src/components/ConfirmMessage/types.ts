export interface IConfirmMessage {
  success: string;
  error: string;
}
