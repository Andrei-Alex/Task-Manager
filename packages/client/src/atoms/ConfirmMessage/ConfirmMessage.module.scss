@import '../../styles/variables';

.success {
  color: green;
}
.error {
color: red;
}
