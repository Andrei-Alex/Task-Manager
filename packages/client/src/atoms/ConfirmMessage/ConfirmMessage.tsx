import React from "react";
import { styles, IConfirmMessage } from "./index";
/**
 *
 * The ConfirmMessage component is a React functional component designed to display confirmation messages in a user interface.
 * It receives two optional props, `success` and `error`, and renders the appropriate message based on the provided prop.
 * The component is memoized using `React.memo` to optimize rendering performance.
 *
 * ## Usage
 * ```jsx
 *   <ConfirmMessage success={successMsg} error={errorMsg}/>
 * ```
 *
 *
 * @component ConfirmMessage
 * @param {Object} props - The props object containing the following properties:
 * @param {string} [props.success] - A string that represents the success message to be displayed.
 * @param {string} [props.error] - A string that represents the error message to be displayed.
 *
 * @returns {JSX.Element|null} - Returns JSX element containing the success or error message, or null if neither prop is provided.
 *
 *
 */


export const ConfirmMessage: React.FC<IConfirmMessage> = ({ success, error }) => {
  return (
    <div>
      {success ? (
        <p className={styles.success}>{success}</p>
      ) : error ? (
        <p className={styles.error}>{error}</p>
      ) : null}
    </div>
  );
};
export default React.memo(ConfirmMessage);
