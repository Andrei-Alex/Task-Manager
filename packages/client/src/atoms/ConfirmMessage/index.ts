export * from "./types";
export { default as styles } from "./ConfirmMessage.module.scss";
