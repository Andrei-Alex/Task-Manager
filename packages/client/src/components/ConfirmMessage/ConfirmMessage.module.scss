@import '../../styles/_variables';

.success {
  color: green;
}
.error {
color: red;
}
