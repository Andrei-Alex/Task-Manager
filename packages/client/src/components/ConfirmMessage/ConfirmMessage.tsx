import React from "react";
import { styles, IConfirmMessage } from "./index";
const ConfirmMessage: React.FC<IConfirmMessage> = ({ success, error }) => {
  return (
    <div>
      {success ? (
        <p className={styles.success}>{success}</p>
      ) : error ? (
        <p className={styles.error}>{error}</p>
      ) : null}
    </div>
  );
};
export default ConfirmMessage;
