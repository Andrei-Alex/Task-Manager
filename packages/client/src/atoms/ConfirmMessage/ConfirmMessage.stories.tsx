import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import { ConfirmMessage } from "./ConfirmMessage";

const meta: Meta<typeof ConfirmMessage> = {
  component: ConfirmMessage,
  title: "Atoms/ConfirmMessage",
  tags: ["autodocs"],
  argTypes: {
    success: {
      name: "Success",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "",
      description: "Display success message",
      control: {
        type: "text",
      },
    },
    error: {
      name: "Error",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "",
      description: "Display error message",
      control: {
        type: "text",
      },
    },
  },
};
export default meta;
type Story = StoryObj<typeof ConfirmMessage>;

export const Success: Story = {
  args: {
    success:"Success !!!",
    error: null,
  },
};
export const Error: Story = {
  args: {
    success: null,
    error:"Error 404 !!!"
  },
};
