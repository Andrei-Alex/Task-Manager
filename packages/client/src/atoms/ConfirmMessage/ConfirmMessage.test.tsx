import React from "react";
import { render } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import ConfirmMessage from "./ConfirmMessage";
import renderer from "react-test-renderer";

describe("ConfirmMessage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ConfirmMessage success={"Success !!!"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should display Success !!! and have CSS class success", () => {
    render(<ConfirmMessage success={"Success !!!"} />);
    const message = screen.getByText("Success !!!");

    expect(message).toBeInTheDocument();
    expect(message).toHaveClass("success");
  });
  it("should display Bad Credentials and have CSS class error", () => {
    render(<ConfirmMessage error={"Bad Credentials !!!"} />);
    const message = screen.getByText("Bad Credentials !!!");

    expect(message).toBeInTheDocument();
    expect(message).toHaveClass("error");
  });
});
