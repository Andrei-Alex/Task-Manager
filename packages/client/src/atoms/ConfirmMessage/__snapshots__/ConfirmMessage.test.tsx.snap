// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmMessage renders correctly 1`] = `
<div>
  <p
    className="success"
  >
    Success !!!
  </p>
</div>
`;
